packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -826,6 +839,25 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      */
     private delayConnectClientId?: string;
 
+    private ensureNoDataModelChangesCalls = 0;
+
+    /**
+     * Invokes the given callback and expects that no ops are submitted
+     * until execution finishes. If an op is submitted, an error will be raised.
+     *
+     * Can be disabled by feature gate `Fluid.ContainerRuntime.DisableOpReentryCheck`
+     *
+     * @param callback - the callback to be invoked
+     */
+    public ensureNoDataModelChanges(callback: () => void): void {
```

Comment:
It is more useful to have ensureNoDataModelChanges<T>(callback: () => T): T;

Reply:
Ah.. Great point!